server/lib/src/idm/account.rs

```diff
@@ -697,6 +706,48 @@ impl Account {
             }
         }
     }
+
+    pub(crate) fn verify_application_password(
+        &self,
+        application: &Application,
+        cleartext: &str,
+    ) -> Result<Option<LdapBoundToken>, OperationError> {
+        if let Some(v) = self.apps_pwds.get(&application.uuid) {
+            for ap in v.iter() {
+                if ap.password.verify(cleartext).map_err(|e| {
+                    error!(crypto_err = ?e);
+                    e.into()
+                })? {
+                    let session_id = uuid::Uuid::new_v4();
+                    security_info!(
+                        "Starting session {} for {} {}",
+                        session_id,
+                        self.spn,
+                        self.uuid
+                    );
+
+                    return Ok(Some(LdapBoundToken {
+                        spn: self.spn.clone(),
+                        session_id,
+                        effective_session: LdapSession::UnixBind(self.uuid),
```

Comment:
Should we consider this being a different type rather than UnixBind since this session Enum is meant to encode "what kind of session this is" and this is an application password session, not a unix session.

Reply:
Makes sense, I added `LdapSession::ApplicationPasswordBind`.